packages/cli-core/src/package-manager-controller/package_manager_controller_base.ts

```diff
@@ -112,7 +112,8 @@ Get started by running \`${this.binaryRunner} amplify sandbox\`.`;
       targetDir,
       this.binaryRunner,
       tscArgs,
-      this.execa
+      this.execa,
+      { preferLocal: true, localDir: targetDir }
```

Comment:
is this the only place impacted?
would this also matter for deployer which runs `tsc` ?

is there something we could do to e2e tests to make them detect this?

Reply:
Good question. I'm wondering that too. I have a commit on my local to improve the e2e, but I need more test before pushing.

Reply:
Could we maybe use `npx which tsc` and verify the path is localDir instead of `/Users/xxxxxx/amplify-backend/node_modules/.bin/tsc`?

Reply:
@rtpascual yea, that's what I did when debugging. Not sure if we need that in the code

Reply:
Just for e2e tests to confirm they are using the correct `tsc`, not sure if there is a better way

Reply:
Now the test should be able to detect if any bin's missing. 

I made commit 99677e753b to delete the .bin of the repo